torchvision/models/detection/generalized_rcnn.py

```diff
@@ -58,8 +58,7 @@ def forward(self, images, targets=None):
 
         """
         if self.training:
-            if targets is None:
-                raise ValueError("In training mode, targets should be passed")
+            torch._assert(targets is not None, "targets should not be none when in training mode")
 
             for target in targets:
```

Comment:
the `if` above made the type checker happy: https://app.circleci.com/pipelines/github/pytorch/vision/16031/workflows/6e480a6f-2735-44f3-972f-dc44e8202d4c/jobs/1297766?invite=true#step-109-23

The problem is that without it the type checker assumes that target can be None in this part and None is not iterable.

I have seen cases that we remove the optional type hint but I would like to see if that's the standard approach here or we have other options other than keeping the exception. 

cc @datumbox